src/ffi.rs

```diff
@@ -93,7 +93,7 @@ impl Code {
     /// an error.
     pub fn jit_compile(&mut self) -> Result<(), Error> {
         let error_code = unsafe {
-            pcre2_jit_compile_8(self.code, PCRE2_JIT_COMPLETE)
+            pcre2_jit_compile_8(self.code, PCRE2_JIT_COMPLETE | PCRE2_JIT_PARTIAL_HARD)
```

Comment:
Does this enable partial matching unconditionally when using the jit? If not, please add a comment and a test verifying that it doesn't. If it does, then we'll need to rethink how we do this because it implies that partial matching isn't just a search-time option, but also a compile-time option.

Reply:
The documentation says:
```
 If you want to use partial matching with just-in-time optimized code, as well as setting a partial match option for the matching function, you must also call pcre2_jit_compile() with one or both of these options:

  PCRE2_JIT_PARTIAL_HARD
  PCRE2_JIT_PARTIAL_SOFT

PCRE2_JIT_COMPLETE should also be set if you are going to run non-partial matches on the same pattern. Separate code is compiled for each mode. If the appropriate JIT mode has not been compiled, interpretive matching code is used. 
```

So I read it, as indeed, this would cause more compile-time overhead, and it is good to give users control over which modes they want to enable.

What about changing this to:

```rust 
pub fn jit_compile(&mut self, &JitOptions options)
```

but then this would break backward compatibility :(
I'm open to ideas here.